packages/doenetml-worker/lib-doenetml-core/src/dast/ref_resolve.rs

```diff
@@ -169,6 +171,56 @@ impl Resolver {
         }
     }
 
+    /// Delete `nodes` from the resolver
+    pub fn delete_nodes(&mut self, nodes: &[FlatNode]) {
+        for node in nodes.iter() {
+            if let FlatNode::Element(element) = node {
+                if let Some(name) = get_element_name(element) {
```

Comment:
We should be able to use a filter/map directly on the `iter` to remove two layers of if statements :-)